jbpm-container-test/jbpm-container-test-suite/src/test/java/org/jbpm/test/container/test/ejbservices/tx/ETransactionTest.java

```diff
@@ -0,0 +1,397 @@
+        } catch (Exception e) {
+            ut.rollback();
+            throw e;
+        }
+
+        ut.commit();
+
+        List<Integer> states = new ArrayList<Integer>();
+        states.add(ProcessInstance.STATE_ABORTED);
+        Collection<ProcessInstanceDesc> processInstances = runtimeDataService.getProcessInstances(states, null,
+                new QueryContext());
+
+        Assertions.assertThat(processInstances).isNotNull().hasSize(1);
+        Assertions.assertThat(processInstances.iterator().next().getId()).isEqualTo(processInstanceId);
+    }
+
+    @Test
+    public void testAbortProcessRollback() throws Exception {
+        Long processInstanceId = startProcessInstance(PROCESS_ID);
+
+        ProcessInstanceDesc processDesc = runtimeDataService.getProcessInstanceById(processInstanceId);
+        Assertions.assertThat(processDesc).isNotNull();
+        Assertions.assertThat(processDesc.getState()).isEqualTo(ProcessInstance.STATE_ACTIVE);
+
+        UserTransaction ut = InitialContext.doLookup(USER_TRANSACTION_NAME);
+        ut.begin();
+
+        try {
+            processService.abortProcessInstance(processInstanceId);
+        } catch (Exception e) {
+            ut.rollback();
+            throw e;
+        }
+        ut.rollback();
+
+        ProcessInstance processInstance = processService.getProcessInstance(processInstanceId);
+        Assertions.assertThat(processInstance).isNotNull();
+        Assertions.assertThat(processInstance.getId()).isEqualTo(processInstanceId);
+        Assertions.assertThat(processInstance.getState()).isEqualTo(ProcessInstance.STATE_ACTIVE);
+    }
+
+    @Test
+    public void testScript() throws Exception {
+        Long processInstanceId = startProcessInstance(PROCESS_ID);
+
+        UserTransaction ut = InitialContext.doLookup(USER_TRANSACTION_NAME);
+        ut.begin();
+
+        try {
+            processService.signalProcessInstance(processInstanceId, "start", "script");
+            Assertions.assertThat(hasNodeLeft(processInstanceId, "script")).isTrue();
+
+        } catch (Exception e) {
+            ut.rollback();
+            throw e;
+        }
+        ut.rollback();
+
+        ut = InitialContext.doLookup(USER_TRANSACTION_NAME);
+        ut.begin();
+
+        try {
+            processService.signalProcessInstance(processInstanceId, "start", "script");
+            Assertions.assertThat(ut.getStatus()).isEqualTo(Status.STATUS_ACTIVE);
+
+        } catch (Exception e) {
+            ut.rollback();
+            throw e;
+        }
+        ut.commit();
+
+        Assertions.assertThat(hasNodeLeft(processInstanceId, "script")).isTrue();
+        processService.signalProcessInstance(processInstanceId, "finish", null);
+        Assertions.assertThat(hasProcessInstanceCompleted(processInstanceId)).isTrue();
+    }
+
+    /*
+     * Support for rules is missing.
+     */
+    @Ignore
+    @Test
+    public void testRuleFlowGroup() throws Exception {
```

Comment:
Is there any Jira for this one? Should there be such tested feature? If not, I would vote for erasing this test.